server/src/main/java/thesistrack/ls1/service/ApplicationService.java

```diff
@@ -0,0 +1,219 @@
+package thesistrack.ls1.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+import thesistrack.ls1.constants.ThesisRoleName;
+import thesistrack.ls1.constants.ThesisState;
+import thesistrack.ls1.controller.payload.LegacyCreateApplicationPayload;
+import thesistrack.ls1.entity.*;
+import thesistrack.ls1.constants.ApplicationState;
+import thesistrack.ls1.exception.request.ResourceInvalidParametersException;
+import thesistrack.ls1.exception.request.ResourceNotFoundException;
+import thesistrack.ls1.repository.ApplicationRepository;
+import thesistrack.ls1.repository.ThesisRepository;
+import thesistrack.ls1.repository.TopicRepository;
+import thesistrack.ls1.repository.UserRepository;
+
+import java.time.Instant;
+import java.util.*;
+
+@Service
+public class ApplicationService {
+    private final ApplicationRepository applicationRepository;
+    private final UserRepository userRepository;
+    private final UploadService uploadService;
+    private final MailingService mailingService;
+    private final TopicRepository topicRepository;
+    private final ThesisRepository thesisRepository;
+    private final ThesisService thesisService;
+
+    @Autowired
+    public ApplicationService(
+            ApplicationRepository applicationRepository,
+            UserRepository userRepository,
+            UploadService storageService,
+            MailingService mailingService,
+            TopicRepository topicRepository, ThesisRepository thesisRepository, ThesisService thesisService) {
+        this.applicationRepository = applicationRepository;
+        this.userRepository = userRepository;
+
+        this.uploadService = storageService;
+        this.mailingService = mailingService;
+        this.topicRepository = topicRepository;
+        this.thesisRepository = thesisRepository;
+        this.thesisService = thesisService;
+    }
+
+    public Page<Application> getAll(String searchString, ApplicationState[] states, int page, int limit, String sortBy, String sortOrder) {
```

Comment:
See my comment above regarding the argument order. Here, the `searchString` (aka `query`) is also flipped with the `state`

Reply:
Makes sense